package/src/components/CartItem/v1/CartItem.js

```diff
@@ -0,0 +1,257 @@
+  static propTypes = {
+    /**
+     * Provided child components to display item data
+     */
+    components: PropTypes.shape({
+      /**
+       * CartItemDetail component
+       */
+      CartItemDetailComponent: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+      /**
+       * Stock warning component
+       */
+      CartItemStockWarningComponent: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+      /**
+       * Price component
+       */
+      CartItemPriceComponent: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+      /**
+       * QuantityInput component
+       */
+      CartItemQuantityInputComponent: PropTypes.oneOfType([PropTypes.string, PropTypes.func])
+    }),
+    /**
+     * CartItem data
+     */
+    item: PropTypes.shape({
+      /**
+       * The cart item ID
+       */
+      _id: PropTypes.string,
+      /**
+       * Array of additional attributes of the chosen item.
+       */
+      attributes: PropTypes.arrayOf(PropTypes.shape({
+        /**
+           * Attribute label (i.e. "Color").
+           */
+        label: PropTypes.string,
+        /**
+           *  Attribute value (i.e. "Red").
+           */
+        value: PropTypes.string
+      })),
+      /**
+       * Current stock quantity of item
+       */
+      currentQuantity: PropTypes.number,
+      /**
+       * Image url of chosen item
+       */
+      imageUrl: PropTypes.string,
+      /**
+       * Price object of chosen item
+       */
+      price: PropTypes.shape({
+        /**
+         * Chosen items compare at price
+         */
+        compareAtPrice: PropTypes.string,
+        /**
+         * Chosen items display price
+         */
+        displayPrice: PropTypes.string
+      }),
+      /**
+       * Chosen items slug
+       */
+      productSlug: PropTypes.string,
+      /**
+       * Chosen items title
+       */
+      title: PropTypes.string,
+      /**
+       * Quantity of chosen item in cart
+       */
+      quantity: PropTypes.number
+    }),
+    /**
+     * On cart item quantity change handler
+     */
+    onChangeCartItemQuantity: PropTypes.func,
+    /**
+     * On remove item from cart handler
+     */
+    onRemoveItemFromCart: PropTypes.func
+  };
+
+  static defaultProps = {
+    components: {
+      CartItemDetailComponent: "Cart Item Detail",
+      CartItemStockWarningComponent: "Cart Item Stock Warning",
+      CartItemPriceComponent: "Cart Item Price",
+      CartItemQuantityInputComponent: "Cart Item Quantity Input"
+    },
+    item: {
+      attributes: [],
+      price: {}
+    },
+    onChangeCartItemQuantity() {},
+    onRemoveItemFromCart() {}
+  };
+
+  state = {
+    isProcessing: false
+  };
+
+  handleChangeCartItemQuantity = (value) => {
+    const { onChangeCartItemQuantity } = this.props;
+    onChangeCartItemQuantity(value);
+  };
+
+  handleRemoveItemFromCart = () => {
+    const { onRemoveItemFromCart } = this.props;
+    onRemoveItemFromCart();
```

Comment:
I believe the the item `_id` needs to be passed to the callback, otherwise, the the parent would not know which item to remone.